src/sync/partial_sync.hpp

```diff
@@ -19,20 +19,35 @@
 #ifndef REALM_OS_PARTIAL_SYNC_HPP
 #define REALM_OS_PARTIAL_SYNC_HPP
 
+#include "impl/results_notifier.hpp"
+
 #include <functional>
 #include <memory>
 #include <string>
 
 namespace realm {
-
 class Realm;
 class Results;
+class Group;
 
 namespace partial_sync {
+	enum class SubscriptionState : int8_t;
+
+	// Returns the default name for subscriptions. Used if a specific name isn't provided.
+	std::string get_default_name(Query query);
```

Comment:
This should not copy the `Query`, as that's quite an expensive operation. It should take the query by reference to avoid this.